internal/testutils/bpffs_windows_test.go

```diff
@@ -0,0 +1,72 @@
+package testutils_test
+
+import (
+	"bytes"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/cilium/ebpf/asm"
+	"github.com/cilium/ebpf/internal"
+	"github.com/cilium/ebpf/internal/sys"
+	"github.com/cilium/ebpf/internal/testutils"
+	"github.com/go-quicktest/qt"
+)
+
+func TestTempBPFFS(t *testing.T) {
```

Comment:
I wonder if this test is really necessary; could we cover this with existing tests that run on both platforms? I get that this is more of a smoke test, but if it wouldn't work correctly, tests in the root package would probably also fail, especially given that t.Cleanup() looks very strict around wanting to remove all pins explicitly.

Reply:
We don't really have tests which pin and then check that the function removes pins, I discovered this the hard way. Mostly because on Linux the cleanup is just `os.RemoveAll`, where here it's more involved. We could make this a test which runs on Linux as well, the benefit there is minimal IMO.

Reply:
That is to say: I think it's worth having this test.